### PR TITLE
Add platform cisco-8000 to platform path

### PR DESCRIPTION
**Pull request by VenkatCisco**
#### Why I did it
Add new platform cisco-8000 directory under sonic-buildimage. 

#### How I did it
create a new directory under platform called "cisco-8000" and introduce rules.mk. 

#### How to verify it

git clone git@github.com:Azure/sonic-buildimage.git
mkdir platform/cisco-8000
<create rules.mk>
make init
make configure PLATFORM=cisco-8000
make all

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [x] 201911
- [x] 202006
- [x] 202012

### CODE REVIEW

File: platform/cisco-8000/rules.mk [hunk 1]
```diff
@@ -0,0 +1,12 @@
+##############################################################
+## Top level platform rules.mk
+##############################################################
+
+CISCO_PLATFORM_BRANCH := cisco-8000/master
+CISCO_PLATFORM_URL := https://github.com/Cisco-8000-sonic/platform-cisco-8000.git
+CISCO_PLATFORM_SRC=$(PLATFORM_PATH)/src
+
+$(shell if [ ! -d $(CISCO_PLATFORM_SRC) ]; then git clone -b $(CISCO_PLATFORM_BRANCH) $(CISCO_PLATFORM_URL) $(CISCO_PLATFORM_SRC); fi)
```
[the remark this answers is not in the document]
Comment (VenkatCisco) [reply platform/cisco-8000/rules.mk:9]: Looks like its failing in which seems to be a breakage of azure/master branch. There is no code intersection. Is there a fix for this issue ?

  File "/sonic/src/sonic-config-engine/tests/test_j2files.py", line 108, in test_zebra_quagga

**Comment by qiluo-msft** [comment]
What is this binary file `platform/cisco-8000/.rules.mk.swp`? Seems a local temp file.

**Comment by VenkatCisco** [comment]
> What is this binary file `platform/cisco-8000/.rules.mk.swp`? Seems a local temp file.

this looks like a file that got added in 'git add' operation. Updated changes are checked in now.

File: platform/cisco-8000/rules.mk (hunk 1 above)
[the remark this answers is not in the document]
Comment (xumia) [reply platform/cisco-8000/rules.mk:9]: It is another issue, build break on master, we can re-build code.

[the remark this answers is not in the document]
Comment (VenkatCisco) [reply platform/cisco-8000/rules.mk:9]: which issue is this ?.

**Comment by VenkatCisco** [comment]
Hi Xuhui,

If the master build breakage issue is fixed, can you help with re-scheduling the run for this PR ?
Thanks so much for the help


Thanks//Venkat

From: xumia ***@***.***>
Reply-To: Azure/sonic-buildimage ***@***.***>
Date: Wednesday, June 9, 2021 at 9:16 PM
To: Azure/sonic-buildimage ***@***.***>
Cc: Venkat Garigipati ***@***.***>, Author ***@***.***>
Subject: Re: [Azure/sonic-buildimage] Add platform cisco-8000 to platform path (#7821)


@xumia commented on this pull request.

________________________________

In platform/cisco-8000/rules.mk<https://github.com/Azure/sonic-buildimage/pull/7821#discussion_r648836476>:

> @@ -0,0 +1,12 @@

+##############################################################

+## Top level platform rules.mk

+##############################################################

+

+CISCO_PLATFORM_BRANCH := cisco-8000/master

+CISCO_PLATFORM_URL := https://github.com/Cisco-8000-sonic/platform-cisco-8000.git

+CISCO_PLATFORM_SRC=$(PLATFORM_PATH)/src

+

+$(shell if [ ! -d $(CISCO_PLATFORM_SRC) ]; then git clone -b $(CISCO_PLATFORM_BRANCH) $(CISCO_PLATFORM_URL) $(CISCO_PLATFORM_SRC); fi)

It is another issue, build break on master, we can re-build code.

File: platform/cisco-8000/rules.mk (hunk 1 above)
[the remark this answers is not in the document]
Comment (xumia) [reply platform/cisco-8000/rules.mk:9]: The PR to fix master issue: https://github.com/Azure/sonic-buildimage/pull/7831

File: platform/cisco-8000/rules.mk [hunk 2]
```diff
@@ -0,0 +1,12 @@
+##############################################################
+## Top level platform rules.mk
+##############################################################
+
+CISCO_PLATFORM_BRANCH := cisco-8000/master
+CISCO_PLATFORM_URL := https://github.com/Cisco-8000-sonic/platform-cisco-8000.git
```
[the remark this answers is not in the document]
Comment (VenkatCisco) [reply platform/cisco-8000/rules.mk:6]: Ack. Please help with creating a private pipeline if that solves your access issues, as cisco would like to keep this access privileges only to MSFT.

File: platform/cisco-8000/rules.mk (hunk 1 above)
[the remark this answers is not in the document]
Comment (VenkatCisco) [reply platform/cisco-8000/rules.mk:9]: All tests were successful. Can you please review & approve this PR

**Comment by VenkatCisco** [comment]
> What is this binary file `platform/cisco-8000/.rules.mk.swp`? Seems a local temp file.
> 
> In reply to: [857422274](https://github.com/Azure/sonic-buildimage/pull/7821#issuecomment-857422274)

this is fix fixed and diffs uploaded.

**Comment by anshuv-mfst** [comment]
6/15: Code upstreaming meeting notes
- please point to commit id instead of branch as @xumia suggested.

**Comment by VenkatCisco** [comment]
platform/cisco-8000 will be leverage the personal access token (PAT) which is used as an alternate to password across Azure DevOps. PAT provides easy way to work with APIs, and is a widespread standard used across well-known organizations and services (Github, GitLab) etc. Instead of using your private credentials which grants full access to ones account, personal access tokens could be a better fit because of their granular permissions. In the long run, PATs are excellent way to scale SONiC across multitude of vendors in case they express fear of compromise of their Intellectual Property. If an external system is compromised, you simply revoke the token instead of changing the password and consequently changing it in all scripts and integrations
 
A detailed description of use Personal Access Token in Azure cloud is @: https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=preview-page
 
Cisco’s use of Personal access tokens for SONIC opens a new set of ways to authenticate into the SONiC environment. Their introduction also serves as a foundational building block for more advanced access control capabilities, including multi-factor authentication and team-based access controls – both areas that we’re working on at the moment. It settles down all concerns of sharing mutually confidential content.
 
Details: 
Cisco will generate a generic username and associated Personal Access Token (PAT) for MSFT’s consumption of cisco-8000. This PAT will be consumed for password during cloning of the cisco-8000 branch. 
 
PAT is acceptable git standard for using tokens instead of passwords via the APIs. Reference for how PAT is generated is @ https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token. PAT is an Azure dev-ops accepted method today 
  
The  username/token will be supplied at ‘make configure’ time using the SONIC_BUILDER_EXTRA_CMDLINE and is specific for PLATFORM=cisco-8000 type.
 
The build commands are : 
make init
make configure PLATFORM=cisco-8000 SONIC_BUILDER_EXTRA_CMDLINE=”-e $GITUSER -e$GITTOKEN”
make all

File: platform/cisco-8000/rules.mk (hunk 1 above)
[the remark this answers is not in the document]
Comment (xumia) [reply platform/cisco-8000/rules.mk:9]: @VenkatCisco , FYI, @liushilongbuaa is working on a feature to make git repo reproducible, when the feature enabled, a commit control file will be generated, we will only download specified commit.

**Review by xumia — APPROVED** [review]

**Review by xumia — APPROVED** [review]

File: platform/cisco-8000/rules.mk [hunk 3]
```diff
@@ -0,0 +1,14 @@
+##############################################################
+## Top level platform rules.mk
+##############################################################
+CISCO_PLATFORM_BRANCH := cisco-8000/master
+CISCO_PLATFORM_URL := https://$(GITUSER):$(GITTOKEN)@Cisco-sonic-8000/platform-cisco-8000.git
```
Comment (xumia) [inline platform/cisco-8000/rules.mk:5]: Does it miss the "github.com/"? I guess the format is as below, correct me if wrong.
https://$(GITUSER):$(GITTOKEN)@github.com/Cisco-sonic-8000/platform-cisco-8000.git

File: platform/cisco-8000/rules.mk [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+CISCO_PLATFORM_SRC=$(PLATFORM_PATH)/src
```
Comment (xumia) [inline platform/cisco-8000/rules.mk:6]: In slave.mk, only include "$(PLATFORM_PATH)/rules.mk", so may need to change the CISCO_PLATFORM_SRC=$(PLATFORM_PATH).
Or another options, you can add platform/cisco-8000/rules.mk, and in the file to include $(PLATFORM_PATH)/src/rules.mk

File: platform/cisco-8000/rules.mk (hunk 3 above)
Comment (xumia) [reply platform/cisco-8000/rules.mk:5]: Cisco-8000-sonic or Cisco-sonic-8000?

File: platform/cisco-8000/rules.mk [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+CISCO_8000_SHA1 := f224715
+
+$(shell if [ ! -d $(CISCO_PLATFORM_SRC) ]; then git clone -b $(CISCO_PLATFORM_BRANCH) $(CISCO_PLATFORM_URL) $(CISCO_PLATFORM_SRC); fi)
+$(shell if [ -d $(CISCO_PLATFORM_SRC) ]; then cd $(CISCO_PLATFORM_SRC) && git checkout $(CISCO_8000_SHA1))
+$(shell if [ -d $(CISCO_PLATFORM_SRC) ]; then $(CISCO_PLATFORM_SRC)/platform_setup.sh; fi)
+
+include $(CISCO_PLATFORM_SRC)/rules.dep
```
Comment (xumia) [inline platform/cisco-8000/rules.mk:13]: rules.mk? See https://github.com/Cisco-8000-sonic/platform-cisco-8000/blob/cisco/201911/rules.mk

**Comment by anshuv-mfst** [comment]
@xumia - can you please help with merge asap, blocking image build.

**Comment by xumia** [comment]
> @xumia - can you please help with merge asap, blocking image build.

I added two comments, it was resolved without any comments.
Could you please help reply it? Please help confirm the repo url you refer to.

CISCO_PLATFORM_BRANCH := cisco-8000/master
CISCO_PLATFORM_URL := https://$(GITUSER):$(GITTOKEN)@Cisco-sonic-8000/platform-cisco-8000.git
CISCO_PLATFORM_SRC=$(PLATFORM_PATH)/src
CISCO_8000_SHA1 := f224715

I cannot access https://github.com/Cisco-sonic-8000, only can access https://github.com/Cisco-8000-sonic/platform-cisco-8000
The URL is in the GITTOKEN?

File: platform/cisco-8000/rules.mk (hunk 1 above)
[the remark this answers is not in the document]
Comment (VenkatCisco) [reply platform/cisco-8000/rules.mk:9]: The PR is significantly updated based on the offline discussion cisco team had with the PR review team from MSFT. There are some more updates checked into the PR for review.

**Comment by VenkatCisco** [comment]
Closing it to track the fix via #8172

**Pull request closed without merging**